qiskit_experiments/measurement/discriminator/discriminator_analysis.py

```diff
@@ -0,0 +1,107 @@
+"""
+Standard discriminator analysis class.
+"""
+
+import numpy as np
+from qiskit_experiments.base_analysis import BaseAnalysis, AnalysisResult
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis, QuadraticDiscriminantAnalysis
+
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class DiscriminatorAnalysis(BaseAnalysis):
+    def _run_analysis(
+        self, experiment_data, discriminator_type="LDA", plot: bool = True, **options
+    ):
+        """Run analysis on discriminator data.
+        Args:
+            experiment_data (ExperimentData): The experiment data to analyze.
+            discriminator_type (str): Type of discriminator to use in analysis. Default is LDA.
+            options: kwarg options for analysis function.
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                ``analysis_results`` may be a single or list of
+                AnalysisResult objects, and ``figures`` may be
+                None, a single figure, or a list of figures.
+        """
+        nqubits = len(experiment_data.data()[0]["metadata"]["ylabel"])
+        discriminator = [None] * nqubits
+        score = [None] * nqubits
+        fig, ax = plt.subplots(nqubits)
+        fig.tight_layout()
+        if nqubits == 1:
+            ax = [ax]
+        for q in range(nqubits):
+            _xdata, _ydata = self._process_data(experiment_data, q)
+
+            if discriminator_type == "LDA":
+                discriminator[q] = LinearDiscriminantAnalysis()
+            elif discriminator_type == "QDA":
+                discriminator[q] = QuadraticDiscriminantAnalysis()
```

Comment:
This should raise an error if `discriminator_type` is something else than `LDA` or `QDA`.